tests/functional/openstackoperator_controller_test.go

```diff
@@ -1065,6 +1065,40 @@ var _ = Describe("OpenStackOperator controller", func() {
 			}, timeout, interval).Should(Succeed())
 		})
 	})
+
+	When("OpenStackControlplane instance is deleted", func() {
+		BeforeEach(func() {
+			DeferCleanup(
+				th.DeleteInstance,
+				CreateOpenStackControlPlane(names.OpenStackControlplaneName, GetDefaultOpenStackControlPlaneSpec()),
+			)
+		})
+
+		It("deletes the OpenStackVersion resource", func() {
+
+			// openstackversion exists
+			Eventually(func(g Gomega) {
+				osversion := GetOpenStackVersion(names.OpenStackControlplaneName)
+				g.Expect(osversion).Should(Not(BeNil()))
+
+				th.ExpectCondition(
+					names.OpenStackControlplaneName,
+					ConditionGetterFunc(OpenStackVersionConditionGetter),
+					corev1.OpenStackVersionInitialized,
+					k8s_corev1.ConditionTrue,
+				)
+			}, timeout, interval).Should(Succeed())
+
+			th.DeleteInstance(GetOpenStackControlPlane(names.OpenStackControlplaneName))
+
+			instance := &corev1.OpenStackVersion{}
+			Eventually(func(g Gomega) {
+				g.Expect(k8sClient.Get(ctx, names.OpenStackControlplaneName, instance)).Should(BeNil()) //nolint:all
```

Comment:
Trying to avoid the nolint here. I tried, Should(Not(Succeed())) but that fails which now makes me think this test isn't quite right. @gibizer if you have time take a look

Reply:
It is a shortcoming of the envtest setup. The automatic garbage collection of k8s is not running in this env. https://book.kubebuilder.io/reference/envtest.html#testing-considerations Checking the OwnerReference and checking that the Finalizer is removed from the Version CR is what we can assert but we cannot assert that the  CR is deleted.

Reply:
bummer. Thanks @gibizer This was a head scratcher. I will adjust it accordingly.